libcasr/src/stacktrace.rs

```diff
@@ -243,13 +243,51 @@ pub fn cluster_stacktraces(stacktraces: &[Stacktrace]) -> Result<Vec<u32>> {
     let mut flat_clusters = vec![0; len];
     for (i, (_, nums)) in clusters.into_iter().enumerate() {
         for num in nums {
-            flat_clusters[num] = i as u32 + 1; // Number clusters from 1, not 0
+            flat_clusters[num] = i + 1; // Number clusters from 1, not 0
         }
     }
 
     Ok(flat_clusters)
 }
 
+/// Perform crashline deduplication for each cluster
+///
+/// # Arguments
+///
+/// * `crashlines` - slice of crashlines as String
+///
+/// * 'clusters' - An vector of the same length as `crashlines`.
+/// Vec\[i\] is the flat cluster number to which original casrep i belongs.
+///
+/// * cluster_num - number of cluster
+///
+/// # Return value
+///
+/// An vector of the same length as `crashlines`.
+/// Vec\[i\] is the flat cluster number to which original casrep i belongs,
+/// if Vec\[i\] is 0, casrep crashline is duplicate of some other
+pub fn dedup_crashlines(
+    crashlines: &[String],
+    clusters: Vec<usize>,
+    cluster_num: usize,
```

Comment:
remove this arg because we can calculate it directly in func

Reply:
Done